runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -163,18 +165,18 @@ public Token appendCheckpoint(Token snapshotTimestamp, Long streamTail) {
             // A checkpoint writer will do two accesses one to obtain the object
             // vlo version and to get a shallow copy of the entry set
             log.info("appendCheckpoint: Started checkpoint for {} at snapshot {}", streamId, snapshotTimestamp);
-            Set<Map.Entry> entries = this.map.entrySet();
+            Stream<Map.Entry> entries = this.map.entryStream();
```

Comment:
`ICorfuSMR` can expose a `getStateMethod`, or `ISMRMap` can expose `getEntryStream`

Reply:
Consumed via `ISMRMap` now.

Reply:
You look at the previous comment regarding `CorfuSMR`.

As for `ICorfuSMR` exposing `getStateMethod` -- I entertained that idea and I concluded that the complexity is not worth the effort. In order to achieve this, we would have to add and extra generic parameter to `ICorfuSMR` `E`, which in case of `CorfuTable` would be `Stream<Map.Entry<K, V>>`, but in other cases it (`CorfuCompoundObject`, `CorfuSharedCounter`) would not be a sequence. Also, after new GC comes in, all of this code is going to be irrelevant.